staging/src/k8s.io/apiserver/pkg/audit/context.go

```diff
@@ -67,35 +71,126 @@ func WithAuditAnnotations(parent context.Context) context.Context {
 // Handlers that are unaware of their position in the overall request flow should
 // prefer AddAuditAnnotation over LogAnnotation to avoid dropping annotations.
 func AddAuditAnnotation(ctx context.Context, key, value string) {
-	// use the audit event directly if we have it
-	if ae := AuditEventFrom(ctx); ae != nil {
-		LogAnnotation(ae, key, value)
+	mutex, ok := auditAnnotationsMutex(ctx)
+	if !ok {
+		klog.ErrorS(nil, "Attempted to add audit annotations from unsupported request chain", "annotation", fmt.Sprintf("%s=%s", key, value))
```

Comment:
if auditing is disabled, this is not registered:

https://github.com/kubernetes/kubernetes/blob/b9792a9daef4d978c5c30b6d10cbcdfa77a9b6ac/staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit_annotations.go#L30-L33

noticed this because this line is getting logged 20k times in integration tests that don't enable auditing

Reply:
opened https://github.com/kubernetes/kubernetes/pull/109181 to fix this